app/admin.go

```diff
@@ -62,20 +58,48 @@ func (a *App) GetLogsSkipSend(page, perPage int) ([]string, *model.AppError) {
 
 		defer file.Close()
 
-		offsetCount := 0
-		limitCount := 0
-		scanner := bufio.NewScanner(file)
-		for scanner.Scan() {
-			if limitCount >= perPage {
-				break
+		var newLine = []byte{'\n'}
+		var lineCount int
+		var searchPos = int64(-1)
+		lineEndPos, err := file.Seek(0, os.SEEK_END)
+		if err != nil {
+			return nil, model.NewAppError("getLogs", "api.admin.file_read_error", nil, err.Error(), http.StatusInternalServerError)
+		}
+		for {
+			pos, err := file.Seek(searchPos, os.SEEK_CUR)
```

Comment:
Sorry totally should have said this before, but os.SEEK_CUR and os.SEEK_END are deprecated:

> Deprecated: Use io.SeekStart, io.SeekCurrent, and io.SeekEnd.

- https://godoc.org/os#pkg-constants

Can you use `io.SeekCurrent` and `io.SeekEnd` instead?